var/spack/repos/builtin/packages/molgw/package.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from os import getenv
+from subprocess import PIPE, run
+
+from spack.package import *
+
+
+class Molgw(MakefilePackage):
+    """MOLGW is a quantum chemistry code that implements the many-body perturbation theory (MBPT)
+    to describe the excited electronic states in finite systems (atoms, molecules, clusters).
+    MOLGW implements the GW approximation for the self-energy (ionization and affinity)
+    and the Bethe-Salpeter equation for the optical excitations.
+    MOLGW also applies the real-time time-dependent density-functional theory (RT-TDDFT).
+    MOLGW employs standard Gaussian basis set.
+    """
+
+    homepage = "https://github.com/bruneval/molgw"
+    url = "https://github.com/bruneval/molgw/archive/v3.2.tar.gz"
+    git = "https://github.com/bruneval/molgw.git"
+
+    maintainers("bruneval")
+
+    # version("rolling-release", branch="master")
+    version("3.2", sha256="a3f9a99db52d95ce03bc3636b5999e6d92b503ec2f4afca33d030480c3e10242")
+
+    # variants
+    variant("openmp", default=False, description="Build with OpenMP support")
+    variant("scalapack", default=False, description="Build with ScaLAPACK support")
+
+    # dependences
+    depends_on("blas")
+    depends_on("lapack")
+    depends_on("libxc@5:")
+    depends_on("libcint+pypzpx+coulomb_erf")
+
+    depends_on("scalapack", when="+scalapack")
+    depends_on("mpi", when="+scalapack")
+
+    # enforce scalapack-capable mkl when asking +scalapack (and using intel-oneapi-mkl)
+    depends_on("intel-oneapi-mkl+cluster", when="+scalapack ^intel-oneapi-mkl")
+    # enforce threaded openblas when asking +openmp (and using openblas)
+    depends_on("openblas threads=openmp", when="+openmp ^openblas")
+
+    def _get_mkl_ld_flags(self, spec):
```

Comment:
This method seems to belong more to some `Intel*` base class, than here. @rscohn2 Do you know if we have anything similar to this?

Reply:
As I wrote in issue #37459, ```intel-oneapi-mkl +cluster threads=openmp``` only works when linking against `intel-mpi`.

See the corresponding lines in `intel-oneapi-mkl/package.py`:
```
        if "+cluster" in self.spec:
            libs.append(self._xlp64_lib("libmkl_blacs_intelmpi"))
```

My coding allows the linking against openmpi.

I agree that in principle the `intel-oneapi-mkl` should be fixed first, but the issue I raised remained unanswered.

Reply:
If you prefer, I can propose a pull request to `intel-oneapi-mkl`. But this package is quite central and I would need some back-up.